src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1008,23 +1007,38 @@ private bool HasHeader
         private readonly IHost _host;
         private const string RegistrationName = "TextLoader";
 
-        public TextLoader(IHostEnvironment env, Column[] columns, Action<Arguments> advancedSettings, IMultiStreamSource dataSample = null)
-            : this(env, MakeArgs(columns, advancedSettings), dataSample)
+        /// <summary>
+        /// Loads a text file into an <see cref="IDataView"/>. Supports basic mapping from input columns to IDataView columns.
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="columns">Defines a mapping between input columns in the file and IDataView columns.</param>
+        /// <param name="hasHeader">Whether the file has a header.</param>
+        /// <param name="separatorChar"> The character used as separator between data points in a row. By default the tab character is used as separator.</param>
+        /// <param name="dataSample">Allows to expose items that can be used for reading.</param>
+        public TextLoader(IHostEnvironment env, Column[] columns, bool hasHeader = false, char separatorChar = '\t', IMultiStreamSource dataSample = null)
+            : this(env, MakeArgs(columns, hasHeader, new[] { separatorChar }), dataSample)
         {
         }
 
-        private static Arguments MakeArgs(Column[] columns, Action<Arguments> advancedSettings)
+        private static Arguments MakeArgs(Column[] columns, bool hasHeader, char[] separatorChars)
         {
-            var result = new Arguments { Column = columns };
-            advancedSettings?.Invoke(result);
+            separatorChars = separatorChars ?? new[] { '\t' };
```

Comment:
(nit) `separatorChars` can never be `null`, right? This is a private method and only called in 1 spot that ensures it won't be null.

Maybe just add an Assert it won't be null and you can remove the null check here.